### PR TITLE
Simplify sample code for {Option, Result}::iter_mut

### LINKED ISSUE #28431: Overcomplicated sample code for Option::iter_mut and Result::iter_mut
The code samples that illustrate the [Option::iter_mut()](https://doc.rust-lang.org/nightly/core/option/enum.Option.html#method.iter_mut) and [Result::iter_mut()](https://doc.rust-lang.org/nightly/core/result/enum.Result.html#method.iter_mut) use an overcomplicated pattern. For instance:

    Some(&mut ref mut v) => *v = 42,

could be simplified to this:

    Some(v) => *v = 42,

or, if we want to make it explicit that the value is a `&mut T`, to this:

    Some(ref mut v) => **v = 42,

but then we need to dereference twice.

In any case, the current sample code seems to bring a lot of confusion for no good reason.

### PR DESCRIPTION
**Pull request by aochagavia**
Fixes #28431

### CODE REVIEW

**Comment by aochagavia** [comment]
cc @steveklabnik

File: src/libcore/result.rs [hunk 1]
```diff
@@ -538,7 +538,7 @@ impl<T, E> Result<T, E> {
     /// ```
     /// let mut x: Result<u32, &str> = Ok(7);
     /// match x.iter_mut().next() {
-    ///     Some(&mut ref mut x) => *x = 40,
+    ///     Some(v) => *v = 40,
```
Comment (aochagavia) [inline src/libcore/result.rs]: I switched from `x` to `v` to avoid shadowing the `x` variable, which could be confusing.

**Comment by bluss** [comment]
Maybe we should use for loops in the examples instead?

**Pull request merged**